### ISSUE TITLE
Why does it need macos and can't run on Linux/qemu?

### ISSUE THREAD

**Issue by christianparpart** [cross-platform-actions/action#8]
Hey.

it's a great effort and many thanks for this work. But while trying to search for a freebsd github action that is _not_ requiring macos for virtualbox but rather Linux with qemu I seem to be out of luck.

I start to wonder why it has to be macos/virtualbox. Isn't qemu (/linux) sufficient? I'm primarily intersted in this because macos costs 10 times the money (or minutes) as standard linux does on github actions pricing.

Many thanks,
Christian.

**Comment by jacob-carlborg** [comment]
The reason why FreeBSD and OpenBSD require macOS runners is because the GitHub action uses the Xhyve hypervisor. Xhyve is only available on macOS. I chose that one because it's faster than QEMU and has minimal dependencies. Also the macOS GitHub runner supports hardware accelerated virtualization while the Linux runners don't. I'm only using this action for open source projects so I haven't thought about the costs.

When running NetBSD, QEMU is already being used and Linux is currently required. Since the action already supports QEMU most work is already done. I'm working on the last part to unify all this that will allow to run FreeBSD and OpenBSD using the QEMU hypervisor on Linux and to run NetBSD using QEMU on macOS.

To summarize: QEMU on Linux is sufficient and I'm working on supporting that.

**Comment by christianparpart** [comment]
Many thanks for the detailed technical reply.

**Comment by christianparpart** [comment]
Make meant thanks for your great work, @jacob-carlborg .

**Comment by jacob-carlborg** [comment]
Thanks. You need to update to version `0.4.0` to run FreeBSD on Linux runners: https://github.com/cross-platform-actions/action/releases/tag/v0.4.0.

**Comment by FGasper** [comment]
@jacob-carlborg Of note: This failed for me w/ OpenBSD: https://github.com/FGasper/p5-JavaScript-QuickJS/actions/runs/2297373821
